Linking failing when building in Ubuntu 22.04

When building the runtime using clang in linux (Ubuntu 22.04) the compilation fails when linking with

```
/usr/bin/ld: Lib/libopenvdb.a(Grid.cc.o): relocation R_X86_64_PC32 against symbol `_ZTVN7openvdb5v10_18KeyErrorE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [CMakeFiles/picogk.dir/build.make:156: lib/picogk.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:185: CMakeFiles/picogk.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

I can fix this by adding the following to CMakeLists.txt (I put it at line 90)

```
#enable position independent code
set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Though I'm not sure if this will break the build for other os's

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,6 +50,15 @@ project(PicoGK VERSION ${LIB_VERSION})
 # Set the C++ standard to C++20 or higher
 set(CMAKE_CXX_STANDARD 20)
 
+# Check for position independent executable support
+include(CheckPIESupported)
+check_pie_supported(OUTPUT_VARIABLE output LANGUAGES C CXX)
+if( NOT CMAKE_C_LINK_PIE_SUPPORTED or NOT CMAKE_CXX_LINK_PIE_SUPPORTED)
+  message(WARNING "PIE is not supported at link time: ${output}.\n"
+                  "PIE link options will not be passed to linker.")
+endif()
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
 # Set the output directory for the generated libraries and executables
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Dist)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/Lib)

